src/clojure/nrepl/server.clj

```diff
@@ -34,12 +49,64 @@
     (catch java.io.IOException e
       (log e "Failed to close " x))))
 
+(defn- default-logger
+  [tag ba]
+  (log/info tag (map byte ba)))
+
+(defn- create-pipes
+  "Create intermediate streams using pipes so you can sniff at input
+  and output socket streams."
+  [^Socket s logger]
+  (let [in (io/input-stream s)
+        out (io/output-stream s)
+        pin-out (PipedOutputStream.)
+        pin-in (PipedInputStream. pin-out)
+        pout-out (PipedOutputStream.)
+        pout-in (PipedInputStream. pout-out)]
+    (future
+      (try
+        (loop []
+          (let [buf (byte-array 8192)
```

Comment:
Btw, is there are any particular reason for this size?

Reply:
Not really, has some number in mind?

Reply:
Ah, no. Just seemed somewhat random and I decided to ask.